src/coreclr/jit/unwindarm.cpp

```diff
@@ -649,8 +679,19 @@ void Compiler::unwindEmitFunc(FuncInfoDsc* func, void* pHotCode, void* pColdCode
     }
 #endif // FEATURE_CFI_SUPPORT
 
-    func->uwi.Allocate((CorJitFuncKind)func->funKind, pHotCode, pColdCode, true);
+    if (func->funKind == FUNC_ROOT)
+    {
+        func->uwi.Allocate((CorJitFuncKind)func->funKind, pHotCode, pColdCode, true);
+    }
+    else
+    {
+        // If the function is split, EH funclets are always cold.
+        const bool isHotCode = (fgFirstColdBlock == nullptr);
```

Comment:
```suggestion
        const bool isHotCode = (func->funKind == FUNC_ROOT) || (fgFirstColdBlock == nullptr);
```
then get rid of the `if` condition.

Reply:
Thanks for catching that. This logic will probably be removed altogether in my next refactor, though.